drivers/py/driver.py

```diff
@@ -122,6 +122,18 @@ def run_driver(unix=False, address="", port=12345, driver=Dummy_driver()):
         elif header == Message("GETFORCE"):
             sock.sendall(Message("FORCEREADY"))
 
+            if not isinstance(forces, np.ndarray) and forces.dtype == np.float64:
+                raise ValueError(
+                    "driver returned forces with the wrong type: we need a "
+                    "numpy.ndarray containing 64-bit floating points values"
+                )
+
+            if not isinstance(vir, np.ndarray) and vir.dtype == np.float64:
+                raise ValueError(
```

Comment:
This could also check for the matrix shape, if the data is contiguous (not sure what the behaviour of send_data would be with non contiguous arrays), and if it is stored in C order.